rtc.go

```diff
@@ -697,8 +698,14 @@ func (r *RTC) onSignalHandle() error {
 			if !payload.Subscription.Success {
 				log.Errorf("suscription error: %v", payload.Subscription.Error)
 			}
+		case *rtc.Reply_Error:
+			if payload.Error.Reason == "goaway" && r.OnGoAway != nil {
```

Comment:
FWIW I think the 302 code is the thing to key on here, not the reason.

Reply:
Ok, submitted another patch https://github.com/dustMason/ion-sdk-go/pull/4, can you take a look?